satpy/readers/vii_utils.py

```diff
@@ -0,0 +1,96 @@
+
+import logging
+import scipy.interpolate as si
+import numpy as np
+
+# PLANCK COEFFICIENTS FOR CALIBRATION AS DEFINED BY EUMETSAT
+C1 = 1.191062e+8   # [W/m2·sr-1·µm4]
+C2 = 1.4387863e+4  # [K·µm]
+
+# CONSTANTS DEFINING THE TIE POINTS
+TIE_POINTS_FACTOR = 8    # Sub-sampling factor of tie points wrt pixel points
+SCAN_ALT_TIE_POINTS = 4  # Number of tie points along the satellite track for each scan
+
+# MEAN EARTH RADIUS AS DEFINED BY IUGG
+MEAN_EARTH_RADIUS = 6371008.7714  # [m]
+
+logger = logging.getLogger(__name__)
+
+
+def tie_points_interpolation(data_on_tie_points):
+    """Interpolate the data from the tie points to the pixel points.
+
+    Args:
+        data_on_tie_points: numpy ndarray containing the values defined on the tie points.
+
+    Returns:
+        ndarray: array containing the interpolated values on the pixel points.
+
+    """
+    # Extract the dimensions of the tie points array across and along track
+    n_tie_act, n_tie_alt = data_on_tie_points.shape
+
+    # Check that the number of tie points along track is multiple of the number of tie points per scan
+    if n_tie_alt % SCAN_ALT_TIE_POINTS != 0:
+        logger.warning("The number of tie points in the along-route dimension must be a multiple of %d",
+                       SCAN_ALT_TIE_POINTS)
+        raise ValueError("The number of tie points in the along-route dimension must be a multiple of %d",
+                         SCAN_ALT_TIE_POINTS)
+
+    # Compute the number of scans
+    n_scans = n_tie_alt // SCAN_ALT_TIE_POINTS
+
+    # Compute the dimensions of the pixel points array across and along track (total and per scan)
+    n_pixel_act = (n_tie_act - 1) * TIE_POINTS_FACTOR
+    n_pixel_alt = n_scans * (SCAN_ALT_TIE_POINTS - 1) * TIE_POINTS_FACTOR
+    n_pixel_alt_per_scan = (SCAN_ALT_TIE_POINTS - 1) * TIE_POINTS_FACTOR
+
+    # Create the output array with the correct dimensions
+    data_on_pixel_points = np.empty((n_pixel_act, n_pixel_alt))
+
+    # Create the array for interpolation
+    tie_grid_act = np.arange(0, n_pixel_act + 1, TIE_POINTS_FACTOR)
+    tie_grid_alt = np.arange(0, n_pixel_alt_per_scan + 1, TIE_POINTS_FACTOR)
+    pixels_grid_act = np.arange(0, n_pixel_act + 1)
+    pixels_grid_alt = np.arange(0, n_pixel_alt_per_scan + 1)
+
+    # Interpolate separately for each scan
+    for i_scan in range(n_scans):
+
+        # Select the tie points for the current scan
+        tie_points_for_scan = range(i_scan * SCAN_ALT_TIE_POINTS, (i_scan + 1) * SCAN_ALT_TIE_POINTS)
+        data_on_tie_points_for_scan = data_on_tie_points[:, tie_points_for_scan]
+
+        # Unwrap the data to remove possible discontinuities between degrees (360->0 or 180->-180)
+        # Note that the values are not wrapped again as they were originally,
+        # therefore the user must explicitly bring the returned values to the desired range if needed
+        data_unwrapped = np.degrees(np.unwrap(np.unwrap(np.radians(data_on_tie_points_for_scan), axis=1), axis=0))
```

Comment:
How does this work around the poles ? we usually convert to cartesian coordinates in these cases.